nixos/modules/services/web-apps/writefreely.nix

```diff
@@ -0,0 +1,118 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.writefreely;
+  configFile = pkgs.writeText "config.ini" (generators.toINI {} cfg.settings);
+  settingsFormat = pkgs.formats.ini {};
+in {
+  options.services.writefreely = {
+    enable = mkEnableOption "WriteFreely, a Markdown-based publishing platform.";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.writefreely;
+      defaultText = literalExpression "pkgs.writefreely";
+      description = lib.mdDoc ''
+        Overridable attribute of the WriteFreely package to use.
+      '';
+    };
```

Comment:
maybe `mkPackageOption` would be even better here. completely forgot that existed.